packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -752,33 +746,24 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             const { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize } = this.evalBlobsAndTrees(snapshot);
             const clientTime = networkTime ? overallTime - networkTime : undefined;
 
-            assert(this._snapshotCacheEntry === undefined, 0x0a7 /* "snapshotCacheEntry already defined!" */);
-            this._snapshotCacheEntry = {
-                file: this.fileEntry,
-                type: "snapshot",
-                key: "",
-            };
-
             // There are some scenarios in ODSP where we cannot cache, trees/latest will explicitly tell us when we cannot cache using an HTTP response header.
             const canCache = response.headers.get("disablebrowsercachingofusercontent") !== "true";
             // There maybe no snapshot - TreesLatest would return just ops.
-            const seqNumber: number = (snapshot.trees && (snapshot.trees[0] as any).sequenceNumber) ?? 0;
+            const sequenceNumber: number = (snapshot.trees && (snapshot.trees[0] as any).sequenceNumber) ?? 0;
             const seqNumberFromOps = snapshot.ops && snapshot.ops.length > 0 ?
                 snapshot.ops[0].sequenceNumber - 1 :
                 undefined;
 
-            if (!Number.isInteger(seqNumber) || seqNumberFromOps !== undefined && seqNumberFromOps !== seqNumber) {
-                this.logger.sendErrorEvent({ eventName: "fetchSnapshotError", seqNumber, seqNumberFromOps });
+            const value: ISnapshotCacheValue = { snapshot, sequenceNumber };
+
+            if (!Number.isInteger(sequenceNumber) || seqNumberFromOps !== undefined && seqNumberFromOps !== sequenceNumber) {
+                this.logger.sendErrorEvent({ eventName: "fetchSnapshotError", sequenceNumber, seqNumberFromOps });
+                value.sequenceNumber = undefined;
             } else if (canCache) {
-                const cacheValue: IPersistedCacheValueWithEpoch = {
-                    value: snapshot,
-                    fluidEpoch: this.epochTracker.fluidEpoch,
-                    version: persistedCacheValueVersion,
-                };
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
                 this.cache.persistedCache.put(
                     this._snapshotCacheEntry,
-                    cacheValue,
-                    seqNumber,
+                    value,
                 );
```

Comment:
>; [](start = 17, length = 1)

rather than floating this promise, just add a .catch, and log an error if it fails. that will make debugging easier if it happens